spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/support/converter/SnsMessageConverter.java

```diff
@@ -77,7 +105,7 @@ public Object fromMessage(Message<?> message, Class<?> targetClass, @Nullable Ob
 				? ((SmartMessageConverter) this.payloadConverter).fromMessage(genericMessage, targetClass,
 						conversionHint)
 				: this.payloadConverter.fromMessage(genericMessage, targetClass);
-		return new SnsMessageWrapper(jsonNode.path("Subject").asText(), convertedMessage);
+		return convertedMessage;
```

Comment:
The reason is because when `NotificationMessageArgumentResolver`  tries to resolving the messages, it might be a single message or a message list. So it will be more tricky to keep the wrapper when processing this situation.

So that's the reason I removed the wrapper.